### ISSUE TITLE
git hack: cannot convert observed branch to feature branch

### ISSUE THREAD

**Issue by ruudk** [git-town/git-town#4176]
### Steps to reproduce

❯ git town propose

[orestis-ruud/phpunit11] git fetch --prune --tags

Error: cannot propose observed branches

❯ git town hack
branch "orestis-ruud/phpunit11" is now a feature branch

❯ git town propose

[orestis-ruud/phpunit11] git fetch --prune --tags

Error: cannot propose observed branches

❯ git town config

Branches:
  main branch: main
  perennial branches: (none)
  perennial regex: (not set)
  default branch type: observed
  feature regex: ^ruudk\/.*
  parked branches: (none)
  contribution branches: (none)
  contribution regex: ^dependabot/
  observed branches: (none)
  observed regex: (none)

### Command output

```shell
.
```

**Comment by ruudk** [comment]
Is it because of my `feature regex` requirement? I would expect this to be used as default, not as a constraint?

**Comment by ruudk** [comment]
That means that the output of this command is weird:

> git town hack
> branch "orestis-ruud/phpunit11" is now a feature branch

If this doesn't do anything, why does it say that it's *now* a feature branch?

I think it makes sense to mark branches differently locally. 

Having force flags everywhere feels like a patch to mask the problem.

**Comment by kevgo** [comment]
Good points, and thanks for sharing a better perspective on this!

**Comment by ruudk** [comment]
Another use case:

For CI reasons, I needed to have a branch that was prefixed with `dependabot/` (some checks run on this branch type).

But git town says Nope:

```command
git town propose
[dependabot/docker/php-84] git fetch --prune --tags

Error: cannot propose contribution branches
```

**Comment by kevgo** [comment]
This is now available as part of [Git Town 17.2](https://github.com/git-town/git-town/releases/tag/v17.2.0). Enjoy!
